src/http/client.cc

```diff
@@ -19,10 +19,116 @@
  * Copyright (C) 2022 Scylladb, Ltd.
  */
 
+#include <seastar/core/loop.hh>
+#include <seastar/core/when_all.hh>
 #include <seastar/http/client.hh>
+#include <seastar/http/request.hh>
+#include <seastar/http/reply.hh>
+#include <seastar/http/response_parser.hh>
+#include <seastar/http/internal/content_source.hh>
 
 namespace seastar {
 namespace http {
 
+client::client(connected_socket&& fd)
+        : _fd(std::move(fd))
+        , _read_buf(_fd.input())
+        , _write_buf(_fd.output())
+{
+}
+
+future<> client::write_body(request& req) {
+    if (req.body_writer) {
+        return req.body_writer(internal::make_http_chunked_output_stream(_write_buf)).then([this] {
+            return _write_buf.write("0\r\n\r\n");
```

Comment:
I don't understand what I'm seeing here. Isn't this last chunk part of what the http_chunked_output_stream thing does when closed? Didn't we need this to work properly even before this patch, for responses, where this last 0 is necessary if you want to reuse the same connection?

Reply:
>  Isn't this last chunk part of what the http_chunked_output_stream thing does when closed?

No, http_chunked_output_stream doesn't send termination chunk.

> Didn't we need this to work properly even before this patch, for responses, where this last 0 is necessary if you want to reuse the same connection?

Httpd code adds this thing explicitly. Maybe it's worth generalizing this part into the chunked stream close method indeed